### PR TITLE
feat: Add shutter sound control

### LINKED ISSUE #311: ✨ How do I enable/disable the shutter sound
### Question

How do I enable or disable the shutter sound when taking a photo?

For some reason, when testing on my physical iPhone device, the sound plays when flash is off but doesn't play when flash is on.

Is this customizable? Thanks!

### What I tried

_No response_

### VisionCamera Version

2.4.1

### Additional information

- [ ] I am using Expo
- [X] I have read the [Troubleshooting Guide](https://mrousavy.github.io/react-native-vision-camera/docs/guides/troubleshooting)
- [X] I agree to follow this project's [Code of Conduct](https://github.com/mrousavy/react-native-vision-camera/blob/main/CODE_OF_CONDUCT.md)
- [X] I searched for similar questions [in the issues page](https://github.com/mrousavy/react-native-vision-camera/issues?q=is%3Aopen+is%3Aissue+label%3A%22❓+question%22) as well as [in the discussions page](https://github.com/mrousavy/react-native-vision-camera/discussions) and found none.

### PR DESCRIPTION
**Pull request by imjihun**
## What
 * This PR Adds shutter sound control

## Changes
 * This PR changes Add disableShutterSound option to takePhoto function (default: false)

## Tested on
 * iPhone 12 Pro, iOS 15.4.1
 * Android 10 API 29 (Galaxy Note9)

## Related issues
 * Fixes #311, #1044

### CODE REVIEW

**Comment by mrousavy** [comment]
Hey, thanks for your PR!

As mentioned in the previous PR (https://github.com/mrousavy/react-native-vision-camera/pull/1072), I'd rather have a prop called `enableShutterSound` that is `true` by default, rather than using reverse logic here. Makes more sense to me.

Also, I think this can be implemented on Android as well, so would be great if we could add that and don't have that limited for iOS only?

File: ios/CameraView+TakePhoto.swift [hunk 1]
```diff
@@ -103,7 +103,13 @@ extension CameraView {
         photoSettings.isAutoContentAwareDistortionCorrectionEnabled = enableAutoDistortionCorrection
       }
 
-      photoOutput.capturePhoto(with: photoSettings, delegate: PhotoCaptureDelegate(promise: promise))
+      // disableShutterSound
+      var disableShutterSound = false
+      if let _disableShutterSound = options["disableShutterSound"] as? Bool{
+        disableShutterSound = _disableShutterSound
+      }
```
Comment (mrousavy) [inline ios/CameraView+TakePhoto.swift]: This probably looks a bit cleaner as a one-liner? not sure though
```suggestion
      let disableShutterSound = (options["disableShutterSound"] as? Bool) ?? false
```

**Comment by imjihun** [comment]
@mrousavy Thanks for the suggestion!

File: android/src/main/java/com/mrousavy/camera/CameraView+TakePhoto.kt [hunk 2]
```diff
@@ -70,6 +72,9 @@ suspend fun CameraView.takePhoto(options: ReadableMap): WritableMap = coroutineS
       Log.d(CameraView.TAG, "Taking picture...")
       val startCapture = System.nanoTime()
       val pic = imageCapture!!.takePicture(takePhotoExecutor)
+      if(!disableShutterSound) {
+        takePhotoSound!!.play(MediaActionSound.SHUTTER_CLICK)
+      }
```
Comment (mrousavy) [inline android/src/main/java/com/mrousavy/camera/CameraView+TakePhoto.kt:77]:
```suggestion
      if (!disableShutterSound) {
        takePhotoSound.play(MediaActionSound.SHUTTER_CLICK)
      }
```

**Comment by mrousavy** [comment]
Hey @imjihun! Thanks for your edit, on Android it looks good. 

But as I said in my first review, I'd want the thing called `enableShutterSound` instead of `disableShutterSound`. Then it is `false` by default, and the user can enable it if he wants. This makes more sense to me than the reverse logic. What do you think?

**Comment by jeffBoom** [comment]
> Hey @imjihun! Thanks for your edit, on Android it looks good.
> 
> But as I said in my first review, I'd want the thing called `enableShutterSound` instead of `disableShutterSound`. Then it is `false` by default, and the user can enable it if he wants. This makes more sense to me than the reverse logic. What do you think?

Personally, I think having the shutter sound on should be the default behavior. I have a need to disable the shutter sound but I would imagine it's not that common?

**Comment by mrousavy** [comment]
I think we should have it disabled by default, that's the default behaviour on both native iOS and native Android :)

**Comment by jeffBoom** [comment]
> I think we should have it disabled by default, that's the default behaviour on both native iOS and native Android :)

Ah my bad I didn't know that. Makes sense.

**Comment by copterisimo** [comment]
@jeffBoom Closely following your PR, looks well done!

**Comment by ueda-tetsuya** [comment]
When will this PR be merged into main?

**Comment by jkaufman** [comment]
Thank you for the PR, I'm glad to see this. Are there plans to merge, @mrousavy? Anything needed from the community to get it over the finish line?

**Comment by mrousavy** [comment]
Hey thanks for your PR but this is fixed in V3!

**Pull request closed without merging**